src/Features/Core/Portable/AddImport/AbstractAddImportCodeFixProvider.cs

```diff
@@ -56,9 +56,19 @@ public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
                 : null;
 
             var installerService = GetPackageInstallerService(document);
-            var packageSources = searchNuGetPackages && symbolSearchService != null && installerService?.IsEnabled(document.Project.Id) == true
-                ? installerService.GetPackageSources()
-                : ImmutableArray<PackageSource>.Empty;
+            ImmutableArray<PackageSource> packageSources;
+            try
+            {
+                // Ensure any issues with getting access to package sources will not break the Code Fix for local imports
+                // https://github.com/dotnet/roslyn/issues/40857
+                packageSources = searchNuGetPackages && symbolSearchService != null && installerService?.IsEnabled(document.Project.Id) == true
+                    ? installerService.GetPackageSources()
+                    : ImmutableArray<PackageSource>.Empty;
+            }
+            catch
```

Comment:
definitely don't want t to just blindly catch here.  Can we at least report the issue through our watson infrastructure in order ot make sure underlying issues are found and fixxed?

Reply:
also, can we extract this out to a `GetPackageSources()` helper?

Reply:
look for `ReportWithoutCrash`

Reply:
Thanks for your input. I found the FatalError.ReportWithoutCrash.

Also while looking into adding a GetPackageSources() helper, I found that there were already two exceptions being caught, that were related to invalid nuget.config. I undid my previous changes and moved the check for the ArgumentException as well as the ReportWithoutCrash there. The other two exceptions at that location seem to be caught for the same reason.